uniref-rest/src/main/java/org/uniprot/api/uniref/service/UniRefLightSearchService.java

```diff
@@ -73,8 +73,14 @@ public UniRefEntryLight findByUniqueId(String uniqueId, String fields) {
         UniRefSearchRequest unirefRequest = (UniRefSearchRequest) request;
         QueryResult<UniRefEntryLight> result = super.search(request);
         if (!unirefRequest.isComplete()) {
-            Stream<UniRefEntryLight> content = result.getContent().map(this::removeOverLimitIds);
-            return QueryResult.of(content, result.getPage(), result.getFacets());
+            Stream<UniRefEntryLight> content =
+                    result.getContent().map(this::removeOverLimitAndCleanMemberId);
+
+            result = QueryResult.of(content, result.getPage(), result.getFacets());
+        } else {
+            Stream<UniRefEntryLight> content = result.getContent().map(this::cleanMemberId);
+
+            result = QueryResult.of(content, result.getPage(), result.getFacets());
         }
         return result;
```

Comment:
we can take out             result = QueryResult.of(content, result.getPage(), result.getFacets());
from if and else blocks.